EIPS/EnsIP01.md

```diff
@@ -0,0 +1,63 @@
+
+Also, not necessarily the initial allocations needs to be the most common way to acquire a domain: if a system like Jim's becomes very popular for secondary markets, then we might be able to see a competition betweeen multiple models, which is good in the long run.
+
+## Overview
+
+The purpose of this document is to explore a way in which the properties of a blind vickrey auction can be kept, while removing the requirement of the user to reveal it a posteriori, by making bids as time-locked messages that can be "mined" by a network of computers. The process would be as follows:
+
+1) **Encoding the bid** The potential bidder decides a name and then creates a random private key, with a small amount of bytes according to a **difficulty factor** given by the contract. Then it uses this small key to encrypt a message containing the name it wants and the amount it is willing to pay for it.
+
+2) **Bidding** the bidder then makes a transaction to the chain with an amount of ether equal to or larger than their secret price. The contract doesn't need to store the full message, only a hash of it, the rest of the message can either be given off-chain to "miners" or it can be logged into an event on-chain to ensure it's available for everyone. The bidder also must reveal the size of the key they picked.
+
+3) **Unlocking** anyone is able to "reveal" the bid by revealing the private key. It can be done by the bidder themselves or by other machines by bruteforcing them. Since the owner has revealed the size of the key, machines can decide not to try to unlock a message if they believe it will not be able to do so
+
+4) **difficulty adjustments** If a key is found in less than 48 hours, the **difficulty** factor is slightly increased, in order to target a median of 48 hours. And in the same manner, whenever a bid is revealed after 48h the difficulty should also decrease ever so slightly
+
+5) **Rewards** Rewards are given to the first person to reveal the key, so anyone bruteforcing it should reveal theirs as soon as they find one, or they might lose the chance of a reward. If no one has found any key after 48h, then the reward should increase exponentially, until a maximum period of 120 hours. After that period, it can only be revealed by the owner, and if the auction has been won by someone else, the bid will be declared invalid. If a large amount of bids go unrevealed, then the rewards should also adjust automatically.
+
+6) **Minimum deposits** The current system has a minimum deposit hard coded to 0.01. This was never considered a final solution because of the floating value of ether and demand for ENS names. In this method the minimum deposit would be proportional to the revealer's fees. A value of twice the reveal fee would be suggested.
```

Comment:
Shouldn't this system be built on top of the existing system? If so, it won't have any input into base-layer parameters like minimum deposit. Personally, I'm a fan of Vlad's proposal to charge rent, and eliminate the minimum fee.

Reply:
I already wrote a proposal about rent and will be publishing it tomorrow